### PR TITLE
Add readonly - handle disabled

### PR DESCRIPTION
**Pull request by zsarnett**
Adds the properties for read-only and disabling the handle. 

https://discordapp.com/channels/330944238910963714/351047592588869643/689488842520395878

### CODE REVIEW

**Comment by zsarnett** [comment]
image

@thomasloven  Can you help me with this SVG issue 

```html
<round-slider value="50" arcLength="180" handleDisabled=true readonly=true startAngle="180" style="--round-slider-path-width: 30px; --round-slider-linecap: square"></round-slider>
```

per Thomas: linecap of square adds square at end. Butt is what is needed here

**Comment by zsarnett** [comment]
@thomasloven What is the status on this? All good if you are busy, just wondering as this is a start to the gauge rewrite. Also, feel free to scrap this PR if you want to build it out the way you want.

**Comment by thomasloven** [comment]
Come to think of it, what does this solve that `handleSize=0` does not?

**Comment by thomasloven** [comment]
Never mind. With `handleSize=0` you can still use the keyboard to adjust the slider.

File: src/main.js [hunk 1]
```diff
@@ -212,6 +217,8 @@ class RoundSlider extends LitElement {
   }
 
   firstUpdated() {
+    if(!this._enabled || this.disabled) return;
+
```
Comment (thomasloven) [inline src/main.js]: With this, the `disabled` property cannot be removed.

File: src/main.js [hunk 2]
```diff
@@ -355,6 +359,9 @@ class RoundSlider extends LitElement {
       .bar {
         stroke: var(--round-slider-bar-color, deepskyblue);
       }
+      svg[disabled] .bar {
+        stroke: var(--round-slider-disabled-path-color, darkgray);
```
Comment (thomasloven) [inline src/main.js]: Should be `--round-slider-disabled-bar-color` for consistency.

File: src/main.js (hunk 1 above)
Comment (zsarnett) [reply src/main.js]: Im not sure I understand what you mean by cannot be removed

Comment (thomasloven) [reply src/main.js]:
```js
constructor() {
  this.disabled = true;
}
firstUpdated() {
  this.disabled = false;
}
render() {
<round-slider
?disabled=${this.disabled}
></round-slider>
}
```

The event listeners will never be registered, and the slider won't work.

Comment (zsarnett) [reply src/main.js]: Ah, gotcha. Good catch. Added this to the updated. Also, updated the `_enabled` name more convey what it is really doing

File: src/main.js [hunk 3]
```diff
@@ -236,9 +233,26 @@ class RoundSlider extends LitElement {
...
+      else if((!this._showHandle || this.disabled) && this.attachedListeners) {
+        document.removeEventListener('mouseup', this.dragEnd.bind(this));
+        document.removeEventListener('touchend', this.dragEnd.bind(this), {passive: false});
+        document.removeEventListener('mousemove', this.drag.bind(this));
+        document.removeEventListener('touchmove', this.drag.bind(this), {passive: false});
+        document.removeEventListener('keydown', this._keyStep.bind(this));
+        this.attachedListeners = false;
+      }
+    }
+  }
```
Comment (thomasloven) [inline src/main.js]: Why not just return early from the listeners if they're not needed?
Am I missing something?
Will it affect how events pass through the element?

Comment (zsarnett) [reply src/main.js]: Hmm we can definitely do that but I would think its better for performance if we don't need to be listening to these events then we shouldn't be

Comment (thomasloven) [reply src/main.js]: I'll just trust you on that one.

Comment (zsarnett) [reply src/main.js]: I mean I am no expert :) We can ask the expert Bram/Paulus. I would be curious if it actually is better to remove them. Or if it is the same

Comment (thomasloven) [reply src/main.js]: Doesn't matter, because this doesn't even work... 😝

File: src/main.js [hunk 4]
```diff
@@ -117,7 +120,9 @@ class RoundSlider extends LitElement {
   }
 
   dragStart(ev) {
-    let handle = ev.target;
+    if(!this._showHandle || this.disabled) return;
+
+    let handle = ev.target;
```
Comment (thomasloven) [inline src/main.js]: Trailing whitespace.

**Pull request merged**